### ISSUE TITLE
speech.isListening works late in iOS simulator and device 

### ISSUE THREAD

**Issue by alihuseyin** [csdcorp/speech_to_text#345]
After I press start button it starts listening but speech.isListening method works late. Even in the sample example when you press start button that microphone animation starts late.

**Comment by sowens-csd** [comment]
How long a delay are you seeing between pressing the button and the listening starting? What platform are you testing on?

**Comment by alihuseyin** [comment]
Android Studio Dolphin | 2021.3.1
Xcode 14.0.1
Simulator iPhone 14 pro max iOS 16.0
Real Device iPhone 6 plus

Between 1-2 sec delay.

**Comment by sowens-csd** [comment]
Yes, I'm seeing a similar delay on a real device, less than a second, but not much less. I have previously seen faster response, though there is always a delay interacting with the speech subsystem. I'm not sure whether some of the safety measures I've added to ensure that the audio system is properly reset so that other plugins can use it have added a delay or if new iOS versions have increased the delay. I'll do some more tracing to see if I can track down where the delay comes from but this part of the initialization process is a bit fragile at the OS level, so even if I find an issue I may not be able to change it. 

This is one reason it's important to give UI feedback on when the system is actually listening. The delay is longer if you play start/stop sounds, so if you're using those, which the example does, you may want to stop.

**Comment by sowens-csd** [comment]
I added a bit more logging and looked in more detail at the delays. By far the biggest delay is the call to 
```swift
            try self.audioSession.setCategory(AVAudioSession.Category.playAndRecord, options: [.defaultToSpeaker,.allowBluetooth,.allowBluetoothA2DP])
```
This seems to be a common source of frustration with the AVAudioSession. The problem the speech plugin has that isn't as common is that I can't be certain that I own the audio session. Other plugins, like audio playback, often want it. So I've gone to a fair amount of effort to make sure it can be reused. Unfortunately doing that makes the start listen slower. There might be some work arounds I could do for apps that don't want to reuse the sound system but I don't know how often that would be a useful scenario. One option would be to add an explicit `release` call that apps could make when they want to share the sound subsystem. That does complicate using the plugin though. 

Anyhow, I'll think about it, if you have any ideas let me know. Right now there doesn't seem to be anything easily changed that will speed it up.

**Comment by alihuseyin** [comment]
For now I removed  m4r files in assets/sounds directory. It works much better after removing those files. 
If i want to put back those m4r files in order to hear start and stop sounds i create a boolean isListening and whenever i press start button i do isListening = true , basically i replace speechProvider.isListening with the boolean i created. I know it is not the right way but for now in may case it works.
